src/EventLogExpert.Eventing/Readers/EventLogReader.cs

```diff
@@ -26,10 +26,13 @@ public void Dispose()
         GC.SuppressFinalize(this);
     }
 
-    // BatchSize can cause some weird behavior if it's too large
-    // Tested 1024 which returned failure way too early, 512 caused weird memory bloat
-    // and there was barely a noticeable speed difference between 64 and 512
-    public bool TryGetEvents(out EventRecord[] events, int batchSize = 64)
+    // Pre-Windows 11, a batch being returned can be maximum of (2 MB of data, batchSize count of events).
```

Comment:
Technically one of the last Windows 10 updates, but I don't remember which one and it's not super important because we shouldn't be checking OS version numbers in the first place (and there's no feature checks for this fix).